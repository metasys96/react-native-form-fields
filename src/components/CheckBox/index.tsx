import React, {useState} from 'react';
import {View, Text, TouchableOpacity,StyleSheet,StyleProp,ViewStyle,TextStyle } from 'react-native';
import {Icon} from 'react-native-elements';

interface AppProps {
  isChecked?: boolean;
  containerBgColor?: string;
  containerFlexDirection?: string;
  containerStyle?: any;
  iconContainerStyle?: StyleProp<ViewStyle>;
  labelContainerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  label?: string;
  labelFontSize?: number;
  labelColor?: string;
  checkedIcon?: checkedIcon;
  uncheckedIcon?: uncheckedIcon;
  checkedElement?: boolean;
  uncheckedElement?: boolean;
  onChange?: (e)=>void;
}

const CheckBox = (props: AppProps) => {
  const {
    isChecked,
    label,
    labelFontSize,
    labelColor,
    checkedIcon,
    uncheckedIcon,
    checkedElement,
    uncheckedElement,
    containerBgColor,
    containerFlexDirection,
    containerStyle,
    iconContainerStyle,
    labelContainerStyle,
    labelStyle,
    onChange,
  } = props;

  const [isCheck, setIsChecked] = useState(isChecked);

  const checked = () => {
    setIsChecked(!isCheck);
    if ('function' === typeof onChange) {
      onChange(!isCheck);
    }
  };

  return (
    <TouchableOpacity
      onPress={checked}
      activeOpacity={1}
      style={[
        [DefaultStyles.containerStyle,{
          flexDirection: containerFlexDirection
            ? containerFlexDirection
            : 'row',
          backgroundColor: containerBgColor,
        }],
        containerStyle,
      ]}>
      <View
        style={[
          DefaultStyles.iconContainerStyle,
          iconContainerStyle,
        ]}>
        {isCheck ? (
          checkedElement ? (
            checkedElement
          ) : (
           
            <Icon
              size={checkedIcon.size}
              name={checkedIcon.name}
              type={checkedIcon.type}
              color={checkedIcon.color}
            />
          )
        ) : uncheckedElement ? (
          uncheckedElement
        ) : (
          <Icon
            size={uncheckedIcon.size}
            name={uncheckedIcon.name}
            type={uncheckedIcon.type}
            color={uncheckedIcon.color}
          />
        )}
      </View>
      <View style={[{justifyContent: 'flex-end'}, labelContainerStyle]}>
        {label ? (
          <Text
            style={[{fontSize: labelFontSize, color: labelColor}, labelStyle]}>
            {label}
          </Text>
        ) : null}
      </View>
    </TouchableOpacity>
  );
};
const DefaultStyles = StyleSheet.create({
  iconContainerStyle: {
    justifyContent: 'flex-start',
    height: 20,
    shadowColor: 'rgba(0,0,0, .2)',
    shadowOffset: {height: 1, width: 1},
    shadowOpacity: 1,
    shadowRadius: 1,
  },
  containerStyle :{
    display: 'flex',
    alignItems: 'center',
    height: 40
  }
})

CheckBox.defaultProps = {
  isCheck: false,
  label: '',
  labelColor: 'black',
  labelFontSize: 16,
  containerFlxDirection: 'row',
  checkedIcon: {
    size: 20,
    name: 'check-box',
    type: 'MaterialIcons',
    color: 'black',
  },
  uncheckedIcon: {
    size: 20,
    name: 'check-box-outline-blank',
    type: 'MaterialIcons',
    color: 'black',
  },
};

export default CheckBox;
