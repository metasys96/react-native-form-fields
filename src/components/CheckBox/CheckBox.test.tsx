import {shallow, mount} from 'enzyme';
import CheckBox from './index';
import React from 'react';
import 'react-native';
import {cleanup, fireEvent, render} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import {View, Text, TouchableOpacity} from 'react-native';

afterEach(cleanup);

function getShallowComponent() {
  return shallow(<CheckBox label="test label" />);
}

function renderComponent(onPressFunction: any) {
  return render(<CheckBox label="test label" />);
}

function getJSONRenderer() {
  return renderer.create(<CheckBox />).toJSON();
}

test('create snapshot', () => {
  const component = getJSONRenderer();
  expect(component).toMatchSnapshot();
});

it('renders component', () => {
  const component = getJSONRenderer();
  expect(component).toBeTruthy();
});

it('should contain TouchableOpacity and text as same as length of the options', () => {
  const component = getShallowComponent();
  expect(component.find('TouchableOpacity')).toHaveLength(1);
});

it('set checked the given checkbox', () => {
  const onPressFunction = jest.fn();
  const wrapper = shallow(
    <CheckBox onChange={onPressFunction} label="test label" />,
  );
  wrapper
    .find('TouchableOpacity')
    .first()
    .props()
    .onPress();
  expect(onPressFunction).toHaveBeenCalledTimes(1);
});
