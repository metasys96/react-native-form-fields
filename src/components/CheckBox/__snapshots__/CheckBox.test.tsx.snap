
exports[`create snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "row",
      "height": 40,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 20,
          "justifyContent": "flex-start",
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        underlayColor="white"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 20,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ï„±
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "justifyContent": "flex-end",
        },
        undefined,
      ]
    }
  />
</View>
`;
